service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -325,50 +321,31 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 	}
 	scheduleID := token.GetScheduleId()
 
-	workflowExecution := commonpb.WorkflowExecution{
-		WorkflowId: token.GetWorkflowId(),
-		RunId:      token.GetRunId(),
-	}
-
-	weContext, release, err := handler.historyCache.GetOrCreateWorkflowExecution(
+	workflowContext, err := handler.historyEngine.workflowConsistencyChecker.GetWorkflowContext(
 		ctx,
-		namespaceID,
-		workflowExecution,
-		workflow.CallerTypeAPI,
+		token.Clock,
+		func(mutableState workflow.MutableState) bool {
+			_, ok := mutableState.GetWorkflowTaskInfo(scheduleID)
+			if !ok && scheduleID >= mutableState.GetNextEventID() {
+				handler.metricsClient.IncCounter(metrics.HistoryRespondWorkflowTaskCompletedScope, metrics.StaleMutableStateCounter)
+				return false
+			}
+			return true
+		},
+		definition.NewWorkflowKey(
+			namespaceID.String(),
+			token.WorkflowId,
+			token.RunId,
+		),
 	)
 	if err != nil {
 		return nil, err
 	}
-	defer func() { release(retError) }()
-
-	var msBuilder workflow.MutableState
-	var currentWorkflowTask *workflow.WorkflowTaskInfo
-	var currentWorkflowTaskRunning bool
-	for attempt := 1; ; attempt++ {
-		msBuilder, err = weContext.LoadWorkflowExecution(ctx)
-		if err != nil {
-			return nil, err
-		}
-		if !msBuilder.IsWorkflowExecutionRunning() {
-			return nil, consts.ErrWorkflowCompleted
-		}
-
-		currentWorkflowTask, currentWorkflowTaskRunning = msBuilder.GetWorkflowTaskInfo(scheduleID)
-
-		// First check to see if cache is still up-to-date as it could potentially have stale workflow execution in
-		// some extreme Cassandra failure cases.
-		if currentWorkflowTaskRunning || scheduleID < msBuilder.GetNextEventID() {
-			break
-		}
```

Comment:
is this logic replaced by the inline predicate at line 327?

Reply:
yes